Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/DirectPushService.cs

```diff
@@ -8,47 +8,47 @@ public class DirectPushService : IPushService
 {
     private readonly IPnsRegistrationRepository _pnsRegistrationRepository;
     private readonly ILogger<DirectPushService> _logger;
-    private readonly CancellationToken _cancellationToken;
 
-    public DirectPushService(IPnsRegistrationRepository pnsRegistrationRepository, ILogger<DirectPushService> logger, CancellationToken cancellationToken)
+    public DirectPushService(IPnsRegistrationRepository pnsRegistrationRepository, ILogger<DirectPushService> logger)
     {
         _pnsRegistrationRepository = pnsRegistrationRepository;
         _logger = logger;
-        _cancellationToken = cancellationToken;
     }
 
-    public async Task SendNotification(IdentityAddress recipient, object notification)
+    public async Task SendNotification(IdentityAddress recipient, object notification, CancellationToken cancellationToken)
     {
-        var registrations = await _pnsRegistrationRepository.FindWithAddress(recipient, _cancellationToken);
+        var registrations = await _pnsRegistrationRepository.FindWithAddress(recipient, cancellationToken, track: true);
```

Comment:
I made a mistake in my comment. This line should not set `track` to `true`, of course